src/gosling-track/gosling-track.ts

```diff
@@ -120,19 +136,34 @@ function GoslingTrack(HGC: any, ...args: any[]): any {
             this.extent = { min: Number.MAX_SAFE_INTEGER, max: Number.MIN_SAFE_INTEGER };
 
             // Graphics for highlighting visual elements under the cursor
-            this.mouseOverGraphics = new HGC.libraries.PIXI.Graphics();
-            this.pMain.addChild(this.mouseOverGraphics);
+            this.pMouseHover = new HGC.libraries.PIXI.Graphics();
+            this.pMouseSelection = new HGC.libraries.PIXI.Graphics();
+            this.pMain.addChild(this.pMouseHover);
+            this.pMain.addChild(this.pMouseSelection);
 
             // Brushes on the color legend
             this.gLegend = HGC.libraries.d3Selection.select(this.context.svgElement).append('g');
 
             // Enable click event
+            this.isBrushActivated = false;
             this.pMask.interactive = true;
+            this.gBrush = HGC.libraries.d3Selection.select(this.context.svgElement).append('g');
+            this.mRangeBrush = new OneDimBrushModel(
+                this.gBrush,
+                HGC.libraries,
+                this,
+                this.options.spec.experimental?.brush
+            );
```

Comment:
Hmm I think I see the issue. Track implements a function that the brush needs to call at somepoint in its lifecycle?

could something like

```typescript
            this.mRangeBrush = new OneDimBrushModel(
                this.gBrush,
                HGC.libraries,
                this.options.spec.experimental?.brush
            );
           this.mRangeBrush.onBrush(this.onRangeBrush.bind(this))
```

Reply:
Ah, `.bind(this)` is exactly what I was looking for 😍